engine/src/main/java/org/terasology/logic/common/RetainComponentsComponent.java

```diff
@@ -0,0 +1,11 @@
+package org.terasology.logic.common;
+
+import com.google.api.client.util.Sets;
+import org.terasology.entitySystem.Component;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class RetainComponentsComponent implements Component {
```

Comment:
Please add Javadoc to explain what this component is intended for. I think you can take parts the PR description for that.